compiler/rustc_middle/src/mir/interpret/queries.rs

```diff
@@ -38,6 +39,13 @@ impl<'tcx> TyCtxt<'tcx> {
         ct: ty::Unevaluated<'tcx>,
         span: Option<Span>,
     ) -> EvalToConstValueResult<'tcx> {
+        // Cannot resolve `Unevaluated` constants that contain inference
+        // variables. We reject those here since `resolve_opt_const_arg`
+        // would fail otherwise
+        if ct.substs.has_infer_types_or_consts() {
+            return Err(ErrorHandled::TooGeneric);
+        }
+
```

Comment:
this never gets triggered, so you can yeet this again

Reply:
Wasn't sure of that. But even if this isn't triggered now, is it still not the correct thing to do? We do want to stall in `const_eval_resolve` if we ever found infer vars, no?

Reply:
You could add a `bug!` here I guess, but this should never trigger afaik